include/aikido/robot/ConcreteRobot.hpp

```diff
@@ -0,0 +1,266 @@
+      double timelimit,
+      std::size_t maxNumTrials);
+
+  /// TODO: Replace this with Problem interface.
+  /// Returns a Trajectory that moves the configuration of the metakeleton such
+  /// that the specified bodynode is set to a sample in a goal TSR and
+  /// the trajectory is constrained to a constraint TSR
+  /// \param[in] stateSpace The StateSpace for the metaskeleton
+  /// \param[in] metaSkeleton The MetaSkeleton to plan with.
+  /// \param[in] bodyNode Bodynode whose frame is meant for TSR
+  /// \param[in] goalTsr The goal TSR to move to
+  /// \param[in] constraintTsr The constraint TSR for the trajectory
+  /// \param[in] collisionFree Collision constraint
+  /// \param[in] timelimit Max time (seconds) to spend per planning to each IK
+  /// \return Trajectory to a sample in TSR, or nullptr if planning fails.
+  aikido::trajectory::TrajectoryPtr planToTSRwithTrajectoryConstraint(
+      const aikido::statespace::dart::MetaSkeletonStateSpacePtr& stateSpace,
+      const dart::dynamics::MetaSkeletonPtr& metaSkeleton,
+      const dart::dynamics::BodyNodePtr& bodyNode,
+      const aikido::constraint::dart::TSRPtr& goalTsr,
+      const aikido::constraint::dart::TSRPtr& constraintTsr,
+      const aikido::constraint::dart::CollisionFreePtr& collisionFree,
+      double timelimit);
+
+  /// Plans to a named configuration.
+  /// \param[in] name Name of the configuration to plan to
+  /// \param[in] collisionFree Collision constraint
+  /// \param[in] timelimit Max time (seconds) to spend
+  /// \return Trajectory to the configuration, or nullptr if planning fails
+  aikido::trajectory::TrajectoryPtr planToNamedConfiguration(
+      const std::string& name,
+      const aikido::constraint::dart::CollisionFreePtr& collisionFree,
+      double timelimit);
+
+  /// TODO: This should be revisited once we have Planner API.
+  /// Sets CRRTPlanner parameters.
+  /// \param[in] crrtParameters CRRT planner parameters
+  void setCRRTPlannerParameters(
+      const util::CRRTPlannerParameters& crrtParameters);
+
+private:
+  // Named Configurations are read from a YAML file
+  using ConfigurationMap
+      = std::unordered_map<std::string, const Eigen::VectorXd>;
+
+  std::unique_ptr<aikido::common::RNG> cloneRNG();
+
+  /// Compute velocity limits from the MetaSkeleton
+  Eigen::VectorXd getVelocityLimits(
+      const dart::dynamics::MetaSkeleton& metaSkeleton) const;
+
+  /// Compute acceleration limits from the MetaSkeleton
+  Eigen::VectorXd getAccelerationLimits(
+      const dart::dynamics::MetaSkeleton& metaSkeleton) const;
+
+  /// If this robot belongs to another (Composite)Robot,
+  /// mRootRobot is the topmost robot containing this robot.
+  Robot* mRootRobot;
+
+  /// Name of this robot
+  std::string mName;
+
+  dart::dynamics::MetaSkeletonPtr mRobot;
+  statespace::dart::MetaSkeletonStateSpacePtr mStateSpace;
+
+  // Skeleton containing mRobot
+  dart::dynamics::SkeletonPtr mParentRobot;
```

Comment:
I think it's somewhat confusing to refer to both the `[Meta]Skeleton` and the `Robot` as a robot. Can we call these `mParentSkeleton` or `mParentRobotSkeleton` (or something else)?